### PR TITLE
Some additional methods for streams

### PR DESCRIPTION
**Pull request by jftsang**
I have found these useful for working with key-value pairs, dictionaries and Pandas dataframes.

### CODE REVIEW

File: streamerate/streams.py [hunk 1]
```diff
@@ -590,10 +592,81 @@ def flatMap(self, predicate: Callable[[_K], Iterable[_V]] = _IDENTITY_FUNC) -> "
             return stream(ItrFromFunc(lambda: itertools.chain.from_iterable(self)))
         return stream(ItrFromFunc(lambda: itertools.chain.from_iterable(self.map(predicate))))
 
+    def makeMapping(
```
Comment (asuiu) [inline streamerate/streams.py:595]: `makeMapping`: consider using `pairWith` to highlight that each element is paired with the result of applying `f`

File: streamerate/streams.py [hunk 2]
```diff
@@ -590,10 +592,81 @@ def flatMap(self, predicate: Callable[[_K], Iterable[_V]] = _IDENTITY_FUNC) -> "
...
+        self: "stream[_K]", f: Callable[[_K], _V]
+    ) -> "stream[Tuple[_K, _V]]":
+        """Apply a unary function to a stream of values x and produce a
+        stream of tuples (x, f(x)). Useful for chaining with `.toMap()`.
+        """
+        return self.map(lambda x: (x, f(x)))
+
+    def makeInverseMapping(
```
Comment (asuiu) [inline streamerate/streams.py:603]: `makeInverseMapping`: consider using `pairBy`, which is concise and mirrors `pairWith`, indicating that the function is applied to produce the first element (key) in each tuple.

File: streamerate/tests/test_Streams.py [hunk 3]
```diff
@@ -1428,6 +1482,12 @@ def test_to_list(self):
         s = stream(range(3))
         self.assertListEqual(s.to_list(), [0, 1, 2])
 
+    def test_to_dataframe(self):
+        self.skipTest(
+            "Not implemented, to avoid making Pandas a dependency of "
```
Comment (asuiu) [inline streamerate/tests/test_Streams.py:1487]: This is okay to implement, as the Pandas will be a dependency just of the tests, not of the library, so feel free to include the pandas as dependency to the `pyproject.toml` -> `[tool.poetry.dev-dependencies]`

**Review by asuiu — CHANGES_REQUESTED** [review]
Looks good overall to me, I suggest just a few minor changes.

**Comment by jftsang** [comment]
Thanks for the feedback!

**Review by asuiu — APPROVED** [review]
LGTM, Ship it!
```
              |    |    |                 
             )_)  )_)  )_)              
            )___))___))___)\            
           )____)____)_____)\\
         _____|____|____|____\\\__
---------\                   /---------
  ^^^^^ ^^^^^^^^^^^^^^^^^^^^^
    ^^^^      ^^^^     ^^^    ^^
         ^^^^      ^^^
```

**Comment by asuiu** [comment]
@jftsang 
I've compiled a new version and uploaded it to PyPi: https://pypi.org/project/streamerate/1.1.2/

Many thanks for your contribution!

**Pull request merged**